src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPass.cs

```diff
@@ -35,8 +35,8 @@ internal class CSharpFormattingPass : CSharpFormattingPassBase
             _logger = loggerFactory.CreateLogger<CSharpFormattingPass>();
         }
 
-        // Run after the HTML formatter pass.
-        public override int Order => DefaultOrder - 4;
+        // Run after the HTML and Razor formatter pass.
+        public override int Order => DefaultOrder - 3;
```

Comment:
CPS would use MEF metadata for this, via an `Order` attribute, though that would still just have numbers in it. Roslyn has an ordering concept where you can say "Before X" or "After X" where "X" is some other component.

For the 6 whole formatters that we have, those solutions seem overkill. I'm not sure we should even bother with MEF, we could just something like the below and we can get rid of order, we can get rid of each pass having to check for on type formatting, and it would be super clear:

```cs
if (IsOnType)
{
	new CSharpOnTypeFormattingPass.Execute();
}
else
{
	new HtmlFormattingPass.Execute();
	new RazorFormattingPass.Execute();
	new CSharpFormattingPass.Execute();
}

new ValidateWhitespaceFormattingPass.Execute();
new ValidateDiagnosticsFormattingPass.Execute();
```

I'll log an issue.

Reply:
https://github.com/dotnet/aspnetcore/issues/35078